test/typechecker_tests.erl

```diff
@@ -592,15 +595,184 @@ add_type_pat_test_() ->
+                                          gradualizer:env())),
+
+      ?_assertEqual(?t(#{}),
+                    typechecker:type_diff(?t(#{x => integer()}),
+                                          ?t(#{x := integer()}),
+                                          gradualizer:env())),
+
+      ?_assertEqual(?t(none()),
+                    typechecker:type_diff(?t(#{x => integer()}),
+                                          ?t(#{x => integer()}),
+                                          gradualizer:env())),
+
+      ?_assertEqual(?t(#{x := b}),
+                    typechecker:type_diff(?t(#{x := a|b}),
+                                          ?t(#{x := a}),
+                                          gradualizer:env())),
+
+      ?_assertEqual(?t(#{x => b}),
+                    typechecker:type_diff(?t(#{x => a|b}),
+                                          ?t(#{x := a}),
+                                          gradualizer:env())),
+
+      ?_assertEqual(?t(#{x := 2..3}),
+                    typechecker:type_diff(?t(#{x := 1..3}),
+                                          ?t(#{x := 1}),
+                                          gradualizer:env())),
+
+      ?_assertEqual(?t(none()),
+                    typechecker:type_diff(?t(#{x := a, y := b}),
+                                          ?t(#{x := a, y := b}),
+                                          gradualizer:env())),
+
+      ?_assertEqual(?t(none()),
+                    typechecker:type_diff(?t(#{x := a, y := b}),
+                                          ?t(#{y := b, x := a}),
+                                          gradualizer:env())),
+
+      ?_assertEqual(?t(#{x := b, y := c}),
+                    typechecker:type_diff(?t(#{x := a|b, y := c}),
+                                          ?t(#{x := a, y := c}),
+                                          gradualizer:env())),
+
+      ?_assertEqual(?t(#{x := b, y => c} | #{x := a|b}),
+                    typechecker:type_diff(?t(#{x := a|b, y => c}),
+                                          ?t(#{x := a, y := c}),
+                                          gradualizer:env())),
+
+      ?_assertEqual(?t(#{x := b, y => c|d} | #{x := a|b, y => d}),
+                    typechecker:type_diff(?t(#{x := a|b, y => c|d}),
+                                          ?t(#{x := a, y := c}),
+                                          gradualizer:env())),
+
+      ?_assertEqual(?t(#{x => b, y := c}),
+                    typechecker:type_diff(?t(#{x => a|b, y := c}),
+                                          ?t(#{x := a, y := c}),
+                                          gradualizer:env())),
+
+      ?_assertEqual(?t(#{y => b}),
+                    typechecker:type_diff(?t(#{x => a, y => b}),
+                                          ?t(#{x := a, y => b}),
+                                          gradualizer:env())),
+
+      ?_assertEqual(?t(#{y => a}),
```

Comment:
Shouldn't it be this? Similar to the case just below and also to `type_diff(?t(#{x => integer()}), ?t(#{x => integer()}))`?
```suggestion
      ?_assertEqual(?t(none()),
```

Reply:
I'm convinced 👍